### ISSUE TITLE
[MDEP-528] Plugin descriptor is missing configuration entries

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-dependency-plugin#967]
**[Martin Höller](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=matinh)** opened **[MDEP-528](https://issues.apache.org/jira/browse/MDEP-528?redirect=false)** and commented

The plugin descriptor seem to be missing some \<configuration> entries, which leads to XML errors in IDEs.

According to the documentation (https://maven.apache.org/plugins/maven-dependency-plugin/examples/exclude-dependencies-from-dependency-analysis.html) the analyse goal supports an "ignoredUsedUndeclaredDependencies" parameter. This parameter works as expected when following the instructions on the mentioned documentation page.

However, the IDE reports invalid XML inside the \<configuation>-tag, for the >ignoredUsedUndeclaredDependencies> tag. Looking at the plugin descriptor plugin.xml inside the maven-dependency-plugin-2.10.jar, there is a configuration-entry that is missing some parameters:

```xml
<configuration>
  <analyzer implementation="java.lang.String" default-value="default">${analyzer}</analyzer>
  <baseDir implementation="java.io.File" default-value="${basedir}"/>
  <failOnWarning implementation="boolean" default-value="false">${failOnWarning}</failOnWarning>
  <ignoreNonCompile implementation="boolean" default-value="false">${ignoreNonCompile}</ignoreNonCompile>
  <outputDirectory implementation="java.io.File" default-value="${project.build.directory}"/>
  <outputXML implementation="boolean" default-value="false">${outputXML}</outputXML>
  <project implementation="org.apache.maven.project.MavenProject" default-value="${project}"/>
  <scriptableFlag implementation="java.lang.String" default-value="$$$%%%">${scriptableFlag}</scriptableFlag>
  <scriptableOutput implementation="boolean" default-value="false">${scriptableOutput}</scriptableOutput>
  <skip implementation="boolean" default-value="false">${mdep.analyze.skip}</skip>
  <verbose implementation="boolean" default-value="false">${verbose}</verbose>
</configuration>
```

Seems like this descriptor needs to be updated/re-generated.

---

**Affects:** 2.10

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Checked from trunk with MPLUGIN 3.5: it pretty much seems like that all array parameters are not considered. I think this is a bug in Maven Plugin Tools. [Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte), what do you think?

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

The mentioned `<configuration/>` is used in case there's a default value and/or a property which can be used to set it. You should have a look at the `parameters/>`

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Just tried 3.0.1-SNAPSHOT in Eclipse. Autocompletion works as expected. [Martin Höller](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=matinh), can you clarify?

**Comment by jira-importer** [comment]
**[Martin Höller](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=matinh)** commented

Hm, I had the described problem ever since I reported it (always with version 2.10). Today I tried updating to 3.0.0 and the problem went away. The strange thing, after reverting the changes to the used plugin version, the problem still was gone. It is working now with 2.10 as well with 3.0.0. Maybe the local repository got corrupted somehow and the update to 3.0.0 fixed the problem?

Anyways, currently everything is working for me and I can't reproduce the problem. Thanks for looking into it!

BTW: I'm working with IntelliJ 2017.1 now. At the time I reported this problem, I was working with 2016.x and verified the problem also in Eclipse. Maybe the IDE change had also some influence.
